### PR TITLE
Fix: showDatePicker should have a simple example in the docs

### LINKED ISSUE #156157: showDatePicker should have a simple example in the docs
In the docs for `showDatePicker` there is sample code for a restorable date picker, but no example of a basic date picker use case. https://api.flutter.dev/flutter/material/showDatePicker.html

The fact that in order to save the result of a date picker selection, you have to use the Future returned from `showDatePicker` could be a little tricky for someone new to Flutter, or anyone not fluent in the docs. A basic example showing that flow above the section on state restoration would be useful.

### PR DESCRIPTION
**Pull request by thejitenpatel**
This PR introduces a basic example of how to use the `showDatePicker` function. The purpose of this PR is to simplify the onboarding process for new Flutter developers by providing a straightforward demonstration of handling the asynchronous Future returned by the showDatePicker. This will help users unfamiliar with the intricacies of asynchronous operations in Flutter.

Fixes #156157 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

File: examples/api/lib/material/date_picker/show_date_picker.1.dart
Comment (nate-thegrate) [inline examples/api/lib/material/date_picker/show_date_picker.1.dart:18]: A couple of ideas:
- Configuring the Scaffold here would provide a tiny performance boost (as `setState` calls wouldn't rebuild the Scaffold)
- Adding an AppBar here would make this consistent with other example files.

```suggestion
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(title: const Text('showDatePicker Example')),
        body: const Center(child: DatePickerExample()),
      ),
    );
```

File: examples/api/lib/material/date_picker/show_date_picker.1.dart
Comment (nate-thegrate) [inline examples/api/lib/material/date_picker/show_date_picker.1.dart:30]: (nit: remove empty line)

```suggestion
```

File: examples/api/lib/material/date_picker/show_date_picker.1.dart
Comment (nate-thegrate) [inline examples/api/lib/material/date_picker/show_date_picker.1.dart:53]: (very important suggestion)

```suggestion
          children: <Widget>[
```

File: packages/flutter/lib/src/material/date_picker.dart
Comment (nate-thegrate) [inline packages/flutter/lib/src/material/date_picker.dart:162]: optional suggestion:

```suggestion
/// Tapping the button calls [showDatePicker] and displays the selected date.
```

- The [style guide](https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#canonical-terminology) recommends using "call" over "invoke", so perhaps it should replace "triggered" as well for consistency.
- I slightly prefer "active voice" over "passive voice".

File: examples/api/lib/material/date_picker/show_date_picker.1.dart
Comment (nate-thegrate) [inline examples/api/lib/material/date_picker/show_date_picker.1.dart:33]: A nice thing about stateful widgets is that they have access to the `context` outside the build method!

```suggestion
  Future<void> _selectDate() async {
```

File: examples/api/lib/material/date_picker/show_date_picker.1.dart
Comment (nate-thegrate) [inline examples/api/lib/material/date_picker/show_date_picker.1.dart:63]: If the `context` parameter is removed, this can be a tear-off!

```suggestion
              onPressed: _selectDate,
```

File: examples/api/lib/material/date_picker/show_date_picker.1.dart
Comment (nate-thegrate) [inline examples/api/lib/material/date_picker/show_date_picker.1.dart:40]: We should use pattern-matching here, even though it would make the code more difficult to understand:

```dart
    if (pickedDate case (!= selectedDate)?) {
```

just kidding :)

Comment (nate-thegrate) [reply examples/api/lib/material/date_picker/show_date_picker.1.dart:40]: Probably should add a `mounted` check though, just to set a good example:

```suggestion
    if (mounted && pickedDate != null && pickedDate != selectedDate) {
```

File: examples/api/lib/material/date_picker/show_date_picker.1.dart
Comment (nate-thegrate) [inline examples/api/lib/material/date_picker/show_date_picker.1.dart:50]: (Would be good to remove the Scaffold & Center widgets here, if they're added to the app's build method.)

File: examples/api/lib/material/date_picker/show_date_picker.1.dart
Comment (nate-thegrate) [inline examples/api/lib/material/date_picker/show_date_picker.1.dart:61]: Thanks to https://github.com/flutter/flutter/pull/152890, Rows and Columns now have a convenient `spacing` parameter.

Perhaps it could replace this SizedBox, but it's not a big deal either way 🙂

**Review by nate-thegrate — COMMENTED** [review]
Thanks so much for adding this!

I have a lot of random thoughts & suggestions, so take a look when you have a chance :)

**Comment by thejitenpatel** [comment]
Thanks for the review, @nate-thegrate! I’ve addressed all your suggestions and look forward to any further feedback you may have.

File: examples/api/lib/material/date_picker/show_date_picker.1.dart
Comment (nate-thegrate) [inline examples/api/lib/material/date_picker/show_date_picker.1.dart:61]:
```suggestion
          child: const Text('Select Date'),
```

**Review by nate-thegrate — COMMENTED** [review]
Thank you @thejitenpatel!

At this point I only see 1 small typo (though the change might affect the result of the test's `find.text()` calls).

File: examples/api/lib/material/date_picker/show_date_picker.1.dart
Comment (nate-thegrate) [inline examples/api/lib/material/date_picker/show_date_picker.1.dart:58]:
```suggestion
        Text(
          selectedDate != null
              ? '${selectedDate!.day}/${selectedDate!.month}/${selectedDate!.year}'
              : 'No date selected',
        ),
```

[relevant style guideline](https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#if-you-have-a-newline-after-some-opening-punctuation-match-it-on-the-closing-punctuation)

**Review by nate-thegrate — APPROVED** [review]
LGTM, with a tiny whitespace suggestion.

Thanks so much for contributing!
(We'll be able to merge once the pull request has 2 approvals.)

File: examples/api/lib/material/date_picker/show_date_picker.1.dart
Comment (TahaTesser) [inline examples/api/lib/material/date_picker/show_date_picker.1.dart:46]:
```suggestion
    setState(() {
      selectedDate = pickedDate;
    });
```

This seems way too complicated than neccesary.

File: packages/flutter/lib/src/material/date_picker.dart
Comment (TahaTesser) [inline packages/flutter/lib/src/material/date_picker.dart:161]:
```suggestion
/// Tapping the button calls [showDatePicker] and displays the selected date.
```

It seems little confusing, tapping the button doesn't display selected date in the example


Maybe
`Tapping the button displays a date picker which returns the selected date.`

**Review by TahaTesser — CHANGES_REQUESTED** [review]

**Comment by thejitenpatel** [comment]
Thanks for the review @TahaTesser! I have made the necessary changes, please have a look and let me know if there's anything needs to be done.

File: examples/api/lib/material/date_picker/show_date_picker.1.dart
Comment (nate-thegrate) [reply examples/api/lib/material/date_picker/show_date_picker.1.dart]: Thanks for bringing this up! I can share the if-statement's rationale:

- `mounted`: Flutter currently has a [use_build_context_synchronously](https://dart.dev/lints/use_build_context_synchronously) lint to prevent a situation where a `context` is referenced after being disposed of. There's also a [proposal](https://github.com/dart-lang/linter/issues/4892) to have the same apply to `setState` calls as well.
- `pickedDate != null`: If the user taps "OK", then the future returned by `showDatePicker` completes with the selected DateTime value; otherwise its value is `null`. Including the null check in the if-statement ensures that tapping outside the date picker doesn't remove the previous selection.
- `pickedDate != selectedDate`: If these two are equal, then rebuilding isn't necessary.

The `mounted` check won't ever be false in this example app, but I feel that our documentation should still encourage this coding practice.

Comment (TahaTesser) [reply examples/api/lib/material/date_picker/show_date_picker.1.dart]: 1. Regarding `mounted`, I'm not sure if it used when the example  has no issues that could benefit from it. When someone is reading this sample might get lost what it does without the level experience you've. They might think mounted is necessary.

2. `pickedDate != null`, if the user presses ok without selection, they're effectively not selecting a date so it the `selectedDate` should be reset IMO. This is an example so they can figure out the logic to not reset the date if they want. The goal of the example is to show how the widget works. Adding complex logic that might seems like it is required when using the widget. 

3. `pickedDate != selectedDate`, setState rebuilds is a common Flutter knowledge. Not sure if it is necessary in a simple widget sample.

**Review by TahaTesser — APPROVED** [review]
LGTM

**Pull request merged**
